src/connection.rs

```diff
@@ -74,6 +118,54 @@ impl Write for Connection {
     fn flush(&mut self) -> io::Result<()> {
         match self.stream {
             Stream::TcpStream(ref mut stream) => stream.flush(),
+            Stream::UdpSocket(ref mut stream) => { 
+                //udp header is 8 bytes in the begining of each datagram
+                let mut udp_header: Vec<u8> = Vec::new();
+                udp_header.write_u16::<BigEndian>(stream.request_id)?; // request id to uniquely identify response for this request
+                udp_header.write_u16::<BigEndian>(0)?; // 0 indicates this is the first datagram for this request
+                udp_header.write_u16::<BigEndian>(1)?; // total datagrams in this request (requests can only be 1 datagram long)
+                udp_header.write_u16::<BigEndian>(256)?; // reserved bytes (tested to also work with zero)
```

Comment:
Looks like the memcached protocal sait this reserved bytes must be zero: https://github.com/memcached/memcached/blob/master/doc/protocol.txt#L1192

Reply:
It's nice if we can build a test case, which will send multiple requests in the same time, and see can we identify which response belongs to which request.

Reply:
> Looks like the memcached protocal sait this reserved bytes must be zero: https://github.com/memcached/memcached/blob/master/doc/protocol.txt#L1192

I took it from libmemcached, but will change it to zero.



> It's nice if we can build a test case, which will send multiple requests in the same time, and see can we identify which response belongs to which request.

I think in the current implementaion, we can send one request at a time. But will add a case with multiple clients connecting to the same server at the same time.